pkg/runtime/channels/channels.go

```diff
@@ -107,7 +109,14 @@ func (c *Channels) Refresh() error {
 	log.Debug("Refreshing channels")
 
 	// Create a HTTP channel for external HTTP endpoint invocation
-	pipeline, err := c.buildHTTPPipelineForSpec(c.appHTTPPipelineSpec, "app channel")
+	var pipeline middlehttp.Pipeline
+	var err error
+	if c.appHTTPPipelineSpec != nil {
```

Comment:
This change applies the middlewares to the appHttpPipeline, but we also have a httpPipeline that is defined in pkg/channel/http. A pipeline needs to be applied there too.

Reply:
Yeah , your right I have passed ```expectedPiplineType``` arg to support that .